pinot-spi/src/main/java/org/apache/pinot/spi/stream/PartitionGroupConsumer.java

```diff
@@ -39,4 +58,19 @@
    */
   MessageBatch fetchMessages(StreamPartitionMsgOffset startOffset, StreamPartitionMsgOffset endOffset, int timeoutMs)
       throws TimeoutException;
+
+  /**
+   * Checkpoints the consumption state of the stream partition group in the source
+   *
+   * This is useful in systems that require preserving consumption state on the source in order to resume or replay
```

Comment:
Pinot preserves the state for all streams. The state is in the StreamPartitionMsgOffset. I don't understand this comment.

Reply:
The "systems" referred in this doc is the source streaming system. Not Pinot.